ros/src/computing/perception/detection/packages/road_wizard/nodes/feat_proj/feat_proj.cpp

```diff
@@ -270,44 +277,97 @@ double GetSignalAngleInCameraSystem(double hang, double vang)
 	return ConvertRadianToDegree(signal_pitch_in_cam);   // holizontal angle of camera is represented by pitch
 }  // double GetSignalAngleInCameraSystem()
 
-
-void echoSignals2(ros::Publisher &pub, bool useOpenGLCoord = false)
+Tbl read_csv(const char* filename, int* max_id)
 {
-	int countPoint = 0;
-	autoware_msgs::Signals signalsInFrame;
-
-	/* Get signals on the path if vecter_map_server is enabled */
-	if (g_use_vector_map_server)
-	{
-		vector_map_server::GetSignal service;
-		/* Set server's request */
-		service.request.pose = g_vector_map_client.pose();
-		service.request.waypoints = g_vector_map_client.waypoints();
+  std::ifstream ifs(filename);
+  std::string line;
+  std::getline(ifs, line); // Remove first line
+
+  Tbl tbl;
```

Comment:
what is tbl?

Reply:
`typedef std::vector<std::vector<std::string>> Tbl;`